### ISSUE TITLE
[BUG] -output_uri_replace fails with CSV when running flow inside DHF, and work with MLCP command in terminal

### ISSUE THREAD

**Issue by taatuut** [marklogic/marklogic-data-hub#3271]
Short description of the problem:

[BUG] -output_uri_replace fails with CSV when running flow inside DHF, and work with MLCP command in terminal.? DHF 5.0.2 / ML 10.0-1.

What behavior are you expecting?

Replacing pattern with provided string for option output_uri_replace

What are the steps to reproduce the issue:

Ingest step for CSV file with following configuration

mlcp.sh import -mode "local" -host "localhost" -port "8010" -username "admin" -password "*****" -input_file_path "/Users/ezegers/Documents/BitBucket/repo/users/ezegers/repos/belastingdienst-mgve/data/custom/cuisine" -input_file_type "delimited_text" -generate_uri "true" -output_collections "IngestCuisine" -output_permissions "rest-reader,read,rest-writer,update" -output_uri_replace "/Users/ezegers/Documents/BitBucket/repo/users/ezegers/repos/belastingdienst-mgve/data/custom/cuisine/,'/data/cuisine/'" -document_type "xml" -transform_module "/data-hub/5/transforms/mlcp-flow-transform.sjs" -transform_namespace "http://marklogic.com/data-hub/mlcp-flow-transform" -transform_param "flow-name=Cuisine,step=1"

Running step from DHF QuickStart does not result in output_uri_replace being applied, when copying command and running in terminal it works as expected.

Which Operating System are you using?
10.14.6 macOS Mojave

Which version of MarkLogic are you using?
10.0.-1

Which version of the Data Hub are you using?
5.0.2

Is this a QuickStart UI bug? If so, which browser are you using?
Google Version 78.0.3904.97 (Official Build) (64-bit)

#### The devil is in the details...

If possible, rerun the command with -d for debugging output and attach the output:

[mldh502debug.txt](https://github.com/marklogic/marklogic-data-hub/files/3864575/mldh502debug.txt)

**Comment by wooldridge** [comment]
@taatuut, thanks for filing this issue, I can reproduce:

1. DH QuickStart v5.0.2
2. Create flow with ingest step for CSV file.
3. Add a replacement setting: /Users/mwooldri/projects/a-replace-test/data/,'/data/'
4. Run flow with ingest step.
5. URIs are not properly replaced, e.g.:
/Users/mwooldri/projects/a-replace-test/data/6ceab56f-0c03-4b6f-9a27-462ff4e018d3.json

On v5.0.4, running through same steps as above. Here I don't see the "Target URI Preview (Read-Only)" field populated correctly when adding the "Target URI Replacement" value.

<img width="1280" alt="replace-test" src="https://user-images.githubusercontent.com/477757/69272424-759ffb80-0b8b-11ea-8345-3198a30fe570.png">

(Running the flow on v5.0.4 yields the same problem, URIs are not properly replaced.)

**Comment by rjrudin** [comment]
I think there's good news in that I couldn't reproduce this on the not-yet-released 5.1.0 baseline. The mlcp command shown by QS was:

```
mlcp.sh import -mode "local" -host "localhost" -port "8010" -username "flow-developer" -password "*****" -input_file_path "/Users/rrudin/dev/workspace/marklogic-data-hub/examples/json-mapping-example/data/csv" -input_file_type "delimited_text" -generate_uri "true" -output_collections "ingestCsv" -output_permissions "rest-reader,read,rest-writer,update" -output_uri_replace "/Users/rrudin/dev/workspace/marklogic-data-hub/examples/json-mapping-example,''" -document_type "json" -transform_module "/data-hub/5/transforms/mlcp-flow-transform.sjs" -transform_namespace "http://marklogic.com/data-hub/mlcp-flow-transform" -transform_param "flow-name=csvingest,step=1"
```

When I run the step via QS, I get the expected 4 JSON docs in staging (I have 4 rows in my CSV file). The URIs are of the form "/data/csv/127b450a-dfe1-4108-a988-94a3d712ef9e.json". 

When I run the step via MLCP, I get the same expected 4 JSON docs, just with URIs of the form "/data/csv/test1.csv-0-1". 

So I don't believe this is an issue any longer in the 5.1 baseline.
